packages/blade/src/components/Carousel/__tests__/Carousel.native.test.tsx

```diff
@@ -0,0 +1,254 @@
+        </CarouselItem>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+      </Carousel>,
+    );
+    expect(queryAllByTestId('active-slide')[0]).toHaveTextContent('0');
+    expect(getByRole('tab', { selected: true })).toHaveProp('slideIndex', 0);
+
+    const [previousButton, nextButton] = queryAllByRole('button');
+    expect(previousButton.props.accessibilityLabel).toBe('Previous Slide');
+    expect(nextButton.props.accessibilityLabel).toBe('Next Slide');
+
+    fireEvent.press(previousButton);
+
+    expect(onChange).toHaveBeenLastCalledWith(3);
+    expect(getByRole('tab', { selected: true })).toHaveProp('slideIndex', 3);
+
+    fireEvent.press(nextButton);
+    expect(onChange).toHaveBeenLastCalledWith(0);
+    expect(getByRole('tab', { selected: true })).toHaveProp('slideIndex', 0);
+  });
+
+  it('should auto play', async () => {
+    jest.useFakeTimers({ legacyFakeTimers: true });
+    const onChange = jest.fn();
+    const { queryAllByTestId, getByRole } = renderWithTheme(
+      <Carousel autoPlay onChange={onChange}>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+      </Carousel>,
+    );
+    expect(queryAllByTestId('active-slide')[0]).toHaveTextContent('0');
+    expect(getByRole('tab', { selected: true })).toHaveProp('slideIndex', 0);
+
+    await act(async () => {
+      jest.advanceTimersByTime(6000);
+    });
+
+    expect(onChange).toHaveBeenLastCalledWith(1);
+    expect(getByRole('tab', { selected: true })).toHaveProp('slideIndex', 1);
+
+    await act(async () => {
+      jest.advanceTimersByTime(6000);
+    });
+
+    expect(onChange).toHaveBeenLastCalledWith(2);
+    expect(getByRole('tab', { selected: true })).toHaveProp('slideIndex', 2);
+
+    await act(async () => {
+      jest.advanceTimersByTime(6000);
+    });
+
+    expect(onChange).toHaveBeenLastCalledWith(0);
+    expect(getByRole('tab', { selected: true })).toHaveProp('slideIndex', 0);
+  });
+
+  it('should not auto play when user is scrolling', async () => {
+    jest.useFakeTimers({ legacyFakeTimers: true });
+
+    const onChange = jest.fn();
+    const { getByLabelText, queryAllByTestId, getByRole } = renderWithTheme(
+      <Carousel accessibilityLabel="My Carousel" autoPlay onChange={onChange}>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+      </Carousel>,
+    );
+
+    expect(queryAllByTestId('active-slide')[0]).toHaveTextContent('0');
+    expect(getByRole('tab', { selected: true })).toHaveProp('slideIndex', 0);
+
+    await act(() => {
```

Comment:
<img width="713" alt="Screenshot 2023-08-11 at 7 58 53 AM" src="https://github.com/razorpay/blade/assets/46647141/2f2a9719-d26a-4350-9c2d-0a8ef9b33147">

```suggestion
    await act(async () => {
```

Reply:
Done.